packages/lib-core/src/extensions/pages.ts

```diff
@@ -0,0 +1,56 @@
+import type { AnyObject } from '@monorepo/common';
+import type { ComponentType } from 'react';
+import type { RouteComponentProps } from 'react-router';
+import type { ExtensionK8sGroupKindModel, ExtensionK8sModel } from '../api/common-types';
+import type { CodeRef, Extension } from '../types/extension';
+
+/** Adds new page to Console router. */
+export type RoutePage = Extension<'core.page/route', RoutePageProperties>;
+
+/** Adds new resource list page to Console router. */
+export type ResourceListPage = Extension<
+  'core.page/resource/list',
+  ResourcePageProperties & AnyObject
```

Comment:
@andrewballantyne FYI the `& AnyObject` was a workaround related to JSON schema generation via [ts-json-schema-generator](https://github.com/vega/ts-json-schema-generator) tool.

I think we should remove this workaround here and on other places.

Reply:
We're using the `AnyObject` in frontend-components repository to consume additional props. Could be usefull in these extension definitionas (if someone implements them they can add custom props as well). But if we want to prevent this let's nuke the usage of `AnyObject` here.

Reply:
I think the default should be to explicitly list all possible properties (mandatory and/or optional) and disallow unknown/custom ones.

There may be use cases where we can allow that, but I think it shouldn't be the default pattern.

Reply:
Let's nuke `& AnyObject` part for now, its original purpose was to make https://github.com/vega/ts-json-schema-generator happy in context of Console plugin SDK.

Reply:
Sure thing!